### PR TITLE
Added support for PSR18

### PR DESCRIPTION
**Pull request by Nyholm**
I know this PR maybe premature for you and we may not be ready to drop PHP5 support. I'm soon about to open the official vote for PSR-18. And I wanted to show how it will effect Guzzle.

See the specification: https://github.com/php-fig/fig-standards/blob/master/proposed/http-client/http-client.md
Read the blog post: https://medium.com/php-fig/the-http-client-psr-9c2535132980

### CODE REVIEW

**Comment by jeremeamia** [comment]
Seems pretty straightforward.

**Comment by Nyholm** [comment]
The voting for the working group is open. @jeremeamia, add your vote here: https://groups.google.com/forum/#!topic/php-fig/BsroPEZcr_M

File: src/Exception/InvalidRequestException.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GuzzleHttp\Exception;
+
+use Psr\Http\Message\RequestInterface;
+
+class InvalidRequestException extends \InvalidArgumentException implements \Psr\Http\Client\Exception\RequestException,  GuzzleException
```
Comment (gmponos) [inline src/Exception/InvalidRequestException.php]: Maybe you could have a look on this https://github.com/guzzle/guzzle/pull/2163. If we could merge this PR you can have this exception extend that one.

**Comment by boesing** [comment]
How about a sattelite project for guzzle to support PSR-18 instead of directly integrating it here?
Something like `guzzlehttp/guzzle-psr18` with just a decorator inside.

Could work like the decorators for zend-cache which provide either PSR-6 or PSR-16 support.
https://github.com/zendframework/zend-cache/tree/master/src/Psr/SimpleCache
https://github.com/zendframework/zend-cache/blob/master/src/Psr/SimpleCache/SimpleCacheDecorator.php

**Comment by Nyholm** [comment]
This is merged in master.

**Pull request closed without merging**